pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -105,6 +105,26 @@ type OperationExecutor interface {
 	// actual state of the world to reflect that.
 	UnmountDevice(deviceToDetach AttachedVolume, actualStateOfWorld ActualStateOfWorldMounterUpdater, mounter mount.Interface) error
 
+	// MapVolume is used when the volumeMode is 'Block'. This method creates a symbolic link to the block device on global map path.
+	// Specifically it will:
+	// * Wait for the device to finish attaching (for attachable volumes only).
+	// * Map volume to global map path using symbolic link (for attachable volumes only).
+	// * Update actual state of world to reflect volume is globally mounted (for
+	//   attachable volumes only).
+	// * Map the volume to the pod specific path using symbolic link.
+	// * Update actual state of world to reflect volume is mounted to the pod
+	//   path.
+	MapVolume(waitForAttachTimeout time.Duration, volumeToMount VolumeToMount, actualStateOfWorld ActualStateOfWorldMounterUpdater) error
+
+	// UnmapVolume unmaps symbolic link to the volume from both the pod specified in
+	// volumeToUnmount and global map path. And then, updates the actual state of the world to reflect that.
+	UnmapVolume(volumeToUnmount MountedVolume, actualStateOfWorld ActualStateOfWorldMounterUpdater) error
```

Comment:
Can we get Map and Unmap methods to be symmetrical?

Reply:
Which part do you intend to make symmetrical? Operations in those method?

Reply:
I mean there's one Map function, and two Unmap functions.

Reply:
UnmapVolume and UnmapDevice has different purpose and also timing which is called from reconciler is different.

- UnmapVolume is called after a volume is unmounted status
- UnmapDevice is called after a volume is detached/unmounted status

Can we combine them?

For the filesystem case, we have
- MountVolume   -> call SetUp(w/o attacher), call attacher.MountDevice and SetUp(w/ attacher)
- UnmountVolume -> call TearDown
- UnmountDevice -> ref count of mount point, call attacher.UnmountDevice and check device status if it is opened

And for the block volume case, we will have;
- MapVolume   -> call SetUpDevice(w/o attacher), call MapDevice and SetUpDevice(w/ attacher)
- UnmapVolume -> call TearDownDevice(remove symlinkc from volume map path) and
                            call UnmapDevice(remove symlinkc from global map path)
- UnmapDevice -> ref count of mount point and check device status if it is opened

Reply:
Ah ok, that makes more sense then.